benchmarks/graphics_pipeline/GraphicsBenchmarkApp.h

```diff
@@ -164,7 +166,51 @@ class GraphicsBenchmarkApp
         std::string name;
     };
 
+    struct SpherePipelineKey
+    {
+        uint8_t ps;
+        uint8_t vs;
+        uint8_t vertexFormat;
+        uint8_t vertexAttributeLayout;
+        bool    enableDepth;
+        bool    enableAlphaBlend;
+
+        static_assert(kAvailablePsShaders.size() < (1 << (8 * sizeof(ps))));
+        static_assert(kAvailableVsShaders.size() < (1 << (8 * sizeof(vs))));
+        static_assert(kAvailableVbFormats.size() < (1 << (8 * sizeof(vertexFormat))));
+        static_assert(kAvailableVertexAttrLayouts.size() < (1 << (8 * sizeof(vertexAttributeLayout))));
+
+        bool operator==(const SpherePipelineKey& rhs) const
+        {
+            return ps == rhs.ps &&
+                   vs == rhs.vs &&
+                   vertexFormat == rhs.vertexFormat &&
+                   vertexAttributeLayout == rhs.vertexAttributeLayout &&
+                   enableDepth == rhs.enableDepth &&
+                   enableAlphaBlend == rhs.enableAlphaBlend;
+        }
+
+        struct Hash
+        {
+            // Not a good hash function, but good enough.
```

Comment:
Did you meant shifting by 8 bits?
Because as is, a pipeline with ps=1, vs=2 collides with ps=0,vs=2

Reply:
Right now, `vs = {0,1}`, so it won't collide unless something went seriously wrong, and if we add more vertex shader variant and forget to update the hash, it will create a few hash collision which won't be terrible.

Reply:
I'd like to use `boost::hash_combine` or something similar instead, although I'm not sure what is the licensing requirement for code snippets. They still have collision, just not obvious.

Reply:
>  if we add more vertex shader variant and forget to update the hash, it will create a few hash collision which won't be terrible.

Doesn't this means we'd use the wrong pipeline, hence don't use the correct vertex shader?

Reply:
> I'd like to use `boost::hash_combine` or something similar instead, although I'm not sure what is the licensing requirement for code snippets. They still have collision, just not obvious.

cppref is CC-BY-SA so I suppose no-no, but I the boost license seems to be OK

Reply:
> > if we add more vertex shader variant and forget to update the hash, it will create a few hash collision which won't be terrible.
> 
> Doesn't this means we'd use the wrong pipeline, hence don't use the correct vertex shader?

That's the job of `operator==`

Reply:
Oh right, of course... 🤦‍♂